modules/compiler/utils/source/pass_functions.cpp

```diff
@@ -73,20 +73,36 @@ uint64_t computeApproximatePrivateMemoryUsage(const llvm::Function &fn) {
   return bytes;
 }
 
-void remapConstantExpr(llvm::ConstantExpr *expr, llvm::Constant *from,
-                       llvm::Constant *to) {
-  llvm::SmallVector<llvm::Constant *, 8> newOps;
-  // iterate through the constant expression and create a vector of old and new
+template <typename ConstantT>
+static llvm::SmallVector<llvm::Constant *> getNewOps(ConstantT *constant,
```

Comment:
Does this really need to be a template - would it not just work with Constant *constant as the first arg?

Reply:
`Constant::getOperand` returns a `Value`, `ConstantExpression::getOperand` and `ConstantArray::getOperand` return a `Constant`, so I made this a template to make it work with `SmallVector<Constant*>`

Reply:
But there's two possibilities here, right? Either there's other types of `Constant` that can have non-constant operands and we won't be able to maintain `SmallVector<Constant*>` anyway, or there isn't and we can just cast `Constant::getOperand`'s result to `Constant*`?

Reply:
Yes you are right, I removed the template and added a `cast`.